app/Http/Controllers/Admin/CommunitiesController.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use App\Models\Community;
+
+class CommunitiesController extends Controller
+{
+    private $community;
+
+    public function __construct(Community $community)
+    {
+        $this->community = $community;
+    }
+
+    public function index()
+    {
+        $all_communities = $this->community->orderBy('updated_at', 'desc')->paginate(5);
+        // paginateでページ割り付けできる！
+
+        return view('admin.communities.index')
+                ->with('all_communities', $all_communities);
+    }
+
+    public function store(Request $request)
+    {
+        $request->validate([
+            'name' => 'required|min:1|max:50|unique:communities,name',
+            'image' => 'mimes:jpeg,jpg,png,gif|max:1048',
+        ]); 
+
+        $this->community->name  = ucwords(strtolower($request->name));
+        // coOKing -> cooking -> Cooking
+        $this->community->image = 'data:image/' . $request->image->extension() . ';base64,' . base64_encode(file_get_contents($request->image));
```

Comment:
I suggest not to upload the image in the table but, upload the file/image in the public folder and only save the name in the image column, adjust also the migration change image column datatype from longtext to varchar 30 (or the possible maximum characters you set with the filename)
```

// Validate the request
        $request->validate([
            'file' => 'required|file|mimes:jpg,jpeg,png,pdf,docx|max:2048',
        ]);

        // Handle the file upload
        if ($request->hasFile('file')) {
            $file = $request->file('file');
            $filename = time() . '_' . $file->getClientOriginalName();
            $file->move(public_path('uploads'), $filename);

            // Save the filename to the database
            $fileModel = new File();
            $fileModel->filename = $filename;
            $fileModel->save();

            return back()->with('success', 'File uploaded successfully');
        }
```